src/g_game.c

```diff
@@ -165,7 +165,6 @@ boolean         padlook = false;
 // killough 4/13/98: Make clock rate adjustable by scale factor
 int             realtic_clock_rate = 100;
 static boolean  doom_weapon_toggles;
-static boolean  full_weapon_cycle;
```

Comment:
Done in https://github.com/fabiangreffrath/woof/pull/2020/commits/dabe3b294fb9f975ac80fcb86fa512472401ba6f

Reply:
I revert it back: https://github.com/fabiangreffrath/woof/pull/2020/commits/7315f6bc8e0dbc25a404eb8f5ea5848f4b8693e9

I don't know what this option does, I don't think we need overly complicated Boom weapon switching for `next/prev`.

Reply:
It applies Vanilla Doom weapon switching restrictions to Boom and later complevels. It was correctly implemented before https://github.com/fabiangreffrath/woof/pull/2020/commits/cb813357569aee2f1d92ef9453467d181bbfaa57, and I merely asked to change its default value to "off" so that it doesn't apply anymore. I asked to keep this option in a prior PR, because I am used to the Vanilla Doom weapon switch logic and expect to get the SSG as the "next" weapon after the pistol and only get the SG as the "prev" weapon from there.

Reply:
So it's vanilla `next/prev` weapon switching for complevels Boom and above?

Reply:
Sure, fine with me. Option names must be unambigious.

Reply:
How about "Vanilla Weapon Switch Logic" or "Vanilla Weapon switch Restrictions"?

Reply:
How about "Vanilla Weapon Carousel"? https://github.com/fabiangreffrath/woof/pull/2020/commits/451530498af54572af867eb5acc58ba8a847e763

I think because of the carousel widget we have an excuse to make it config only.

Reply:
Done here: https://github.com/fabiangreffrath/woof/pull/2020/commits/946b5b8c0b5a1804fc5b080fb871c6097d1a672b

Reply:
I pushed this change: https://github.com/fabiangreffrath/woof/pull/2020/commits/ce0908ca474723b85b34a83d6b2547f8d2be96ae

I don't mind that this is off by default, but there will be players like Fabian who will prefer the vanilla weapon cycle (next weapon skips regular shotgun, etc.) even when using Boom complevel or greater. It's been around for so long it's muscle memory for many people.